pkg/resource/podtemplatespec.go

```diff
@@ -336,13 +336,19 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, proxyLister configl
 		}
 	}
 
+	nodeSelectors := map[string]string{}
+	for k, v := range cr.Spec.NodeSelector {
+		nodeSelectors[k] = v
+	}
+	nodeSelectors["kubernetes.io/os"] = "linux"
```

Comment:
Shouldn't we allow users to override our default?

Reply:
I think it makes sense. I have adjusted it to avoid overwrite.